File: app/src/main/java/com/thanksmister/bitcoin/localtrader/network/services/NotificationService.java
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.thanksmister.bitcoin.localtrader.network.services;

import android.content.Context;

import com.thanksmister.bitcoin.localtrader.network.api.model.Notification;
import com.thanksmister.bitcoin.localtrader.utils.NotificationUtils;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import timber.log.Timber;

@Singleton
public class NotificationService {

    private final NotificationUtils notificationUtils;

    @Inject
    public NotificationService(Context context) {
        this.notificationUtils = new NotificationUtils(context);
    }

    public void createNotifications(List<Notification> notifications) {
        if (notifications.isEmpty())
            return;

        List<Notification> notificationList = new ArrayList<Notification>();
        for (Notification notification : notifications) {
            boolean read = notification.getRead();
            boolean support = notification.getUrl().contains("support");
            boolean feedback = notification.getUrl().contains("feedback");
            if (!read && !feedback && !support) {
                notificationList.add(notification);
            }
        }

        // TODO add to translations!!
        if (notificationList.size() > 1) {
            notificationUtils.createNotification("New notifications", "You have " + notificationList.size() + " new notifications.");
        } else if (notificationList.size() == 1) {
            Notification notification = notificationList.get(0);
            if (notification.getContactId() != null) {
                notificationUtils.createNotification("Trade notification", notification.getMessage());
            } else if (notification.getAdvertisementId() != null) {
                notificationUtils.createNotification("Advertisement notification", notification.getMessage());
            } else {
                notificationUtils.createNotification("Notification", notification.getMessage());
            }
        }
    }

    public void balanceUpdateNotification(String title, String ticker, String message) {
        notificationUtils.createNotification(title, message);
    }
}

File: app/src/main/java/com/thanksmister/bitcoin/localtrader/network/services/NotificationService.kt
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.thanksmister.bitcoin.localtrader.network.services

import android.content.Context
import android.text.TextUtils

import com.thanksmister.bitcoin.localtrader.R
import com.thanksmister.bitcoin.localtrader.persistence.Notification
import com.thanksmister.bitcoin.localtrader.utils.NotificationUtils

import java.util.ArrayList

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class NotificationService @Inject
constructor(context: Context) {

    private val notificationUtils: NotificationUtils = NotificationUtils(context)

    fun createNotifications(notifications: List<Notification>) {
        if (notifications.isEmpty()) {
            return
        }

        val notificationList = ArrayList<Notification>()
        for (notification in notifications) {
            val read = notification.read
            if (!TextUtils.isEmpty(notification.url)) {
                val support = notification.url!!.contains("support")
                val feedback = notification.url!!.contains("feedback")
                if (!read && !feedback && !support) {
                    notificationList.add(notification)
                }
            }
        }

        if (notificationList.size > 1) {
            notificationUtils.createNotification(notificationUtils.getString(R.string.notifications_notification_title), notificationUtils.getString(R.string.notifications_description, notificationList.size))
        } else if (notificationList.size == 1) {
            val notification = notificationList[0]
            if (notification.contactId != null) {
                notificationUtils.createNotification(notificationUtils.getString(R.string.notifications_trade_title), notification.msg)
            } else if (notification.advertisementId != null) {
                notificationUtils.createNotification(notificationUtils.getString(R.string.notifications_advertisement_title), notification.msg)
            } else {
                notificationUtils.createNotification(notificationUtils.getString(R.string.notification_title), notification.msg)
            }
        }
    }

    fun createNotification(title: String, message: String) {
        notificationUtils.createNotification(title, message)
    }
}
